packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/__tests__/quanticCategoryFacet.test.js

```diff
@@ -114,74 +189,152 @@ function mockSuccessfulHeadlessInitialization() {
   };
 }
 
-function cleanup() {
-  // The jsdom instance is shared across test cases in a single file so reset the DOM
-  while (document.body.firstChild) {
-    document.body.removeChild(document.body.firstChild);
-  }
-  jest.clearAllMocks();
-  isInitialized = false;
+function mockErroneousHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element) => {
+    if (element instanceof QuanticCategoryFacet) {
+      element.setInitializationError();
+    }
+  };
+}
+
+function setupEventDispatchTest(eventName) {
+  const handler = (event) => {
+    functionsMocks.eventHandler(event?.detail);
+    document.removeEventListener(eventName, handler);
+  };
+  document.addEventListener(eventName, handler);
 }
```

Comment:
I see this is used multiple places in different test files, I would suggest to move it to a utils file